### PR TITLE
master: Default to false for l_bind_docker_reg_auth

### PR DESCRIPTION
**Pull request by ashcrow**
Default to False, but if the required where is tripped then use registry
auth. Without this default ``l_bind_docker_reg_auth`` may not get set
causing install/upgrades to fail in specific cases.

Found when testing https://github.com/openshift/openshift-ansible/pull/5536

### CODE REVIEW

**Comment by ashcrow** [comment]
@michaelgugino it's possible that this only affects 3.5 as upgrades fail without this set.

**Comment by ashcrow** [comment]
To be clearer, 3.5 to 3.6 upgrade.

**Comment by michaelgugino** [comment]
The defaults should be present in templates in all branches going back to 1.4.

I recently updated some of the upgrade components in another PR that was merged, as this variable wasn't being set properly.

This patch set should be unnecessary.  If there is a specific failure you're seeing around this variable, please let me know so I can investigate the logic.  I have an outstanding RFE that I need closed this cycle for registry authentication support.

**Comment by ashcrow** [comment]
@michaelgugino 

**Branch**: release-3.6

**HEAD**: ea9731b2fb4dae833ba595d3291c294010a0b271

**Ansible Version**: 2.3.1.0

**Command**:
```
ansible-playbook -vvv playbooks/byo/openshift-cluster/upgrades/v3_6/upgrade_control_plane.yml 
```

```
<snip/>

TASK [Pre-pull master image] ****************************************************************************************************************************************************************************************
task path: openshift-ansible/roles/openshift_master/tasks/systemd_units.yml:20    
Using module file Envs/py3env/lib/python3.6/site-packages/ansible/modules/commands/command.py                                                                                                       
<192.168.124.234> ESTABLISH SSH CONNECTION FOR USER: root                                                 
<192.168.124.234> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/steve/.ansible/cp/%h-%r 192.168.124.234 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<192.168.124.234> (0, b'\n{"changed": true, "end": "2017-09-26 14:12:24.116068", "stdout": "Trying to pull repository registry.access.redhat.com/openshift3/ose ... \\nsha256:89d4b4a298fe1c738ce28c277ac7c14883bf989130ac1d2a54c81110a11d569f: Pulling from registry.access.redhat.com/openshift3/ose\\nbe1e04c6fbff: Already exists\\n449ed11d916a: Already exists\\nea7d7e15b2fc: Already exists\\n843048ee27a7: Already exists\\nDigest: sha256:89d4b4a298fe1c738ce28c277ac7c14883bf989130ac1d2a54c81110a11d569f\\nStatus: Image is up to date for registry.access.redhat.com/openshift3/ose:v3.6.173.0.21-17", "cmd": ["docker", "pull", "openshift3/ose:v3.6.173.0.21-17"], "rc": 0, "start": "2017-09-26 14:12:20.666352", "stderr": "", "delta": "0:00:03.449716", "invocation": {"module_args": {"warn": true, "executable": null, "_uses_shell": false, "_raw_params": "docker pull openshift3/ose:v3.6.173.0.21-17", "removes": null, "creates": null, "chdir": null}}, "warnings": []}\n', b'')
ok: [192.168.124.234] => {                           
    "changed": false,                                
    "cmd": [                                         
        "docker",                                    
        "pull",                                      
        "openshift3/ose:v3.6.173.0.21-17"            
    ],                                               
    "delta": "0:00:03.449716",                       
    "end": "2017-09-26 14:12:24.116068",             
    "invocation": {                                  
        "module_args": {                             
            "_raw_params": "docker pull openshift3/ose:v3.6.173.0.21-17",                                 
            "_uses_shell": false,                    
            "chdir": null,                           
            "creates": null,                         
            "executable": null,                      
            "removes": null,                         
            "warn": true                             
        }                                            
    },                                               
    "rc": 0,                                         
    "start": "2017-09-26 14:12:20.666352"            
}                                                    

STDOUT:                                              

Trying to pull repository registry.access.redhat.com/openshift3/ose ...                                   
sha256:89d4b4a298fe1c738ce28c277ac7c14883bf989130ac1d2a54c81110a11d569f: Pulling from registry.access.redhat.com/openshift3/ose                                                                                      
be1e04c6fbff: Already exists                         
449ed11d916a: Already exists                         
ea7d7e15b2fc: Already exists                         
843048ee27a7: Already exists                         
Digest: sha256:89d4b4a298fe1c738ce28c277ac7c14883bf989130ac1d2a54c81110a11d569f                           
Status: Image is up to date for registry.access.redhat.com/openshift3/ose:v3.6.173.0.21-17                


TASK [Create the systemd unit files] ********************************************************************************************************************************************************************************
task path: openshift-ansible/roles/openshift_master/tasks/systemd_units.yml:28    
Using module file Envs/py3env/lib/python3.6/site-packages/ansible/modules/files/stat.py  
<192.168.124.234> ESTABLISH SSH CONNECTION FOR USER: root                                                 
<192.168.124.234> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/steve/.ansible/cp/%h-%r 192.168.124.234 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<192.168.124.234> (0, b'\n{"invocation": {"module_args": {"checksum_algorithm": "sha1", "get_checksum": true, "follow": true, "checksum_algo": "sha1", "path": "/etc/systemd/system/atomic-openshift-master.service", "get_md5": false, "get_mime": true, "get_attributes": true}}, "stat": {"charset": "us-ascii", "uid": 0, "exists": true, "attr_flags": "", "woth": false, "isreg": true, "device_type": 0, "mtime": 1505241605.054265, "block_size": 4096, "inode": 33635118, "isgid": false, "size": 778, "executable": false, "isuid": false, "readable": true, "version": "1205536013", "pw_name": "root", "gid": 0, "ischr": false, "wusr": true, "writeable": true, "mimetype": "text/plain", "blocks": 8, "xoth": false, "islnk": false, "nlink": 1, "issock": false, "rgrp": true, "gr_name": "root", "path": "/etc/systemd/system/atomic-openshift-master.service", "xusr": false, "atime": 1506449444.7864316, "isdir": false, "ctime": 1505241605.2732654, "isblk": false, "wgrp": false, "checksum": "9b00f0bbd33f12ae8f6ebe3f5977a9ac0941d433", "dev": 64768, "roth": true, "isfifo": false, "mode": "0644", "xgrp": false, "rusr": true, "attributes": []}, "changed": false}\n', b'')
fatal: [192.168.124.234]: FAILED! => {               
    "changed": false,                                
    "failed": true                                   
}                                                    

MSG:                                                 

AnsibleUndefinedVariable: 'l_bind_docker_reg_auth' is undefined     
```

**Comment by michaelgugino** [comment]
@ashcrow Looks like I missed that template.  They should all have '| default(False)' in the templates.  That's the variable is only ever used in the templates.

I would prefer we use '| default(False)' over set_fact to be uniform with all the other templates.

**Comment by ashcrow** [comment]
@michaelgugino that sounds fair. I'll update this PR to do that instead.

**Comment by michaelgugino** [comment]
@ashcrow Thanks.  Looks like we'll need it here too:
roles/openshift_node_upgrade/templates/openshift.docker.node.service

**Comment by ashcrow** [comment]
@michaelgugino OK updating this to be specifically for 3.6 then.

File: roles/openshift_master/templates/master_docker/master.docker.service.j2 [hunk 1]
```diff
@@ -8,7 +8,7 @@ Wants=etcd_container.service
...
+ExecStart=/usr/bin/docker run --rm --privileged --net=host --name {{ openshift.common.service_type }}-master --env-file=/etc/sysconfig/{{ openshift.common.service_type }}-master -v {{ openshift.common.data_dir }}:{{ openshift.common.data_dir }} -v /var/log:/var/log -v /var/run/docker.sock:/var/run/docker.sock -v {{ openshift.common.config_base }}:{{ openshift.common.config_base }} {% if openshift_cloudprovider_kind | default('') != '' -%} -v {{ openshift.common.config_base }}/cloudprovider:{{ openshift.common.config_base}}/cloudprovider {% endif -%} -v /etc/pki:/etc/pki:ro {% if l_bind_docker_reg_auth | default(False) %} -v {{ oreg_auth_credentials_path }}:/root/.docker:ro{% endif %} {{ openshift.master.master_image }}:${IMAGE_VERSION} start master --config=${CONFIG_FILE} $OPTIONS
```
Comment (ashcrow) [inline roles/openshift_master/templates/master_docker/master.docker.service.j2]: Change is defaulting to ``False``:

```
{% if l_bind_docker_reg_auth | default(False) %}
```

**Pull request merged**